language/movie_nl/main.py

```diff
@@ -0,0 +1,366 @@
+    try:
+        content = requests.get(wiki_url).text
+        return content.split('title')[1].split('-')[0].split('>')[1].strip()
+    except:
+        return os.path.basename(wiki_url).replace('_', ' ')
+
+
+def to_entity_json(entity, e_tuple):
+    """Convert the entity info to json."""
+    json_doc = {}
+
+    avg_sentiment = float(e_tuple[0]) / float(e_tuple[1])
+
+    json_doc['wiki_url'] = entity
+    json_doc['name'] = get_wiki_title(entity)
+    json_doc['sentiment'] = float('%.3f' % e_tuple[0])
+    json_doc['avg_sentiment'] = float('%.3f' % avg_sentiment)
+
+    return json.dumps(json_doc)
+
+
+def get_sentiment_entities(service, document):
+    """Compute the overall sentiment volume in the document"""
+
+    try:
+        sentiments, entities = analyze_document(service, document)
+    except HttpError as e:
+        raise e
+
+    sentiments = [sent for sent in sentiments if sent[0] is not None]
+    negative_sentiments = [
+        polarity for polarity, magnitude in sentiments if polarity < 0.0]
+    positive_sentiments = [
+        polarity for polarity, magnitude in sentiments if polarity > 0.0]
+
+    negative = sum(negative_sentiments)
+    positive = sum(positive_sentiments)
+    total = positive + negative
+
+    return (total, entities)
+
+
+def get_sentiment_label(sentiment):
+    """Return the sentiment label based on the sentiment quantity."""
+    if sentiment < 0:
+        return -1
+    elif sentiment > 0:
+        return 1
+    else:
+        return 0
+
+
+def process_movie_reviews(service, reader, sentiment_writer, entity_writer):
+    """Perform some sentiment math and come up with movie review."""
+    collected_entities = {}
+
+    for document in reader:
+        try:
+            sentiment_total, entities = get_sentiment_entities(
+                service, document)
+        except HttpError as e:
+            logging.error("Error in process_movie_reviews {}".format(e.content))
+            continue
+
+        document.label = get_sentiment_label(sentiment_total)
+
+        sentiment_writer.write(
+            to_sentiment_json(
+                document.doc_id,
+                sentiment_total,
+                document.label
+            )
+        )
+
+        sentiment_writer.write('\n')
+
+        for ent in entities:
+            ent_sent, frequency = collected_entities.get(ent, (0, 0))
+            ent_sent += sentiment_total
+            frequency += 1
+
+            collected_entities[ent] = (ent_sent, frequency)
+
+    for entity, e_tuple in collected_entities.items():
+        entity_writer.write(to_entity_json(entity, e_tuple))
+        entity_writer.write('\n')
+
+    sentiment_writer.flush()
+    entity_writer.flush()
+
+
+def document_generator(dir_path_pattern, count=None):
+    """Generator for the input movie documents."""
+    for running_count, item in enumerate(glob.iglob(dir_path_pattern)):
+        if count and running_count >= count:
+            raise StopIteration()
```

Comment:
FYI the more pythonic way to do this is to slice the list being iterated over:

    for item in glob.iglob(dir_name_pattern)[:count]:

If `count` is `None`, it will correctly give you the entire list.